pkg/controllers/workgenerator/controller_integration_test.go

```diff
@@ -1636,7 +1650,440 @@ var _ = Describe("Test Work Generator Controller", func() {
+			if clusterResourceBinding != nil {
+				Expect(k8sClient.Delete(ctx, clusterResourceBinding)).Should(Succeed(), "Failed to delete cluster resource binding")
+			}
+
+			// Delete the member cluster.
+			memberCluster := clusterv1beta1.MemberCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: memberClusterName,
+				},
+			}
+			Expect(k8sClient.Delete(ctx, &memberCluster)).Should(Succeed(), "Failed to delete member cluster")
+		})
+	})
+
+	Context("Refresh status given different apply strategies (ReportDiff -> ClientSideApply)", Ordered, func() {
+		var clusterResourceBinding *placementv1beta1.ClusterResourceBinding
+
+		var diffObservedTimestamp *metav1.Time
+
+		BeforeAll(func() {
+			memberClusterName = "cluster-" + utils.RandStr()
+			memberCluster := clusterv1beta1.MemberCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: memberClusterName,
+				},
+			}
+			Expect(k8sClient.Create(ctx, &memberCluster)).Should(Succeed(), "Failed to create member cluster")
+
+			memberClusterNamespaceName = fmt.Sprintf(utils.NamespaceNameFormat, memberClusterName)
+			memberClusterReservedNS := corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: memberClusterNamespaceName,
+				},
+			}
+			Expect(k8sClient.Create(ctx, &memberClusterReservedNS)).Should(Succeed(), "Failed to create namespace")
+
+			testCRPName = "crp-" + utils.RandStr()
+		})
+
+		It("can create multiple resource snapshots and resource binding", func() {
+			// Create the master resource snapshot.
+			masterResourceSnapshot := generateResourceSnapshot(
+				1, 2, 0,
+				[][]byte{
+					testResourceCRD, testNameSpace,
+				},
+			)
+			Expect(k8sClient.Create(ctx, masterResourceSnapshot)).Should(Succeed(), "Failed to create master resource snapshot")
+
+			secondaryResourceSnapshot := generateResourceSnapshot(
+				1, 2, 1,
+				[][]byte{
+					testResource,
+				},
+			)
+			Expect(k8sClient.Create(ctx, secondaryResourceSnapshot)).Should(Succeed(), "Failed to create secondary resource snapshot")
+
+			resourceBindingSpec := placementv1beta1.ResourceBindingSpec{
+				State:                placementv1beta1.BindingStateBound,
+				ResourceSnapshotName: masterResourceSnapshot.Name,
+				TargetCluster:        memberClusterName,
+				ApplyStrategy: &placementv1beta1.ApplyStrategy{
+					Type: placementv1beta1.ApplyStrategyTypeReportDiff,
+				},
+			}
+			createClusterResourceBinding(&clusterResourceBinding, resourceBindingSpec)
+		})
+
+		It("can update work status", func() {
+			work := retrieveWork(memberClusterNamespaceName, fmt.Sprintf(placementv1beta1.FirstWorkNameFmt, testCRPName))
+
+			diffReportedCond := &metav1.Condition{
+				Type:               string(placementv1beta1.WorkConditionTypeDiffReported),
+				Status:             metav1.ConditionTrue,
+				Reason:             string(workapplier.ManifestProcessingReportDiffResultTypeNoDiffFound),
+				ObservedGeneration: work.GetGeneration(),
+			}
+			refreshWorkStatus(work, nil, nil, diffReportedCond, nil)
+		})
```

Comment:
we won't need this (which simulated a behavior of another controller) if we explicitly set the applied to be false ourself.

Reply:
Hi Ryan! Part of the code tests the status populator, which is why the condition is being set (summarizing all `DiffReported` conditions from work objects).